controllers/gcpcluster_controller.go

```diff
@@ -43,28 +46,38 @@ const (
 
 //counterfeiter:generate . GKEMembershipClient
 type GKEMembershipClient interface {
-	RegisterMembership(ctx context.Context, cluster *capg.GCPCluster, jwks []byte) (*gkehubpb.Membership, error)
+	Register(ctx context.Context, cluster *capg.GCPCluster, jwks []byte) (*gkehubpb.Membership, error)
+	Unregister(ctx context.Context, cluster *capg.GCPCluster) error
 }
 
 // GCPClusterReconciler reconciles a GCPCluster object
 type GCPClusterReconciler struct {
-	client.Client
-	Logger logr.Logger
-
 	MembershipSecretNamespace string
-	GKEMembershipClient       GKEMembershipClient
+
+	runtimeClient       client.Client
+	GKEMembershipClient GKEMembershipClient
+}
+
+func NewGCPClusterReconciler(membershipSecretNamespace string, runtimeClient client.Client, membershipClient GKEMembershipClient) *GCPClusterReconciler {
+	return &GCPClusterReconciler{
+		runtimeClient:             runtimeClient,
```

Comment:
how come we need to store the client on a different field?

Reply:
you mean instead of embedding it?

Reply:
Yes, like we had before.

Reply:
I just don't see the point of it. I don't want the reconciler to implement the `client.Client` interface, I just want it to use a client when reconciling